devmand/gateway/src/devmand/channels/cli/ReadCachingCli.cpp

```diff
@@ -23,18 +25,18 @@ Future<string> devmand::channels::cli::ReadCachingCli::executeAndRead(
     const Command& cmd) {
   const string command = cmd.toString();
   if (!cmd.skipCache()) {
-    string cachedResult = cache->withWLock([=](auto& cache_) {
+    Optional<string> cachedResult = cache->withWLock([command, this](auto& cache_) -> Optional<string> {
```

Comment:
just a nitpick, but if we want to be explicit, we should instead of "this' use "id = this->id"